Allow secrets in `AWS_Credential`

radeusgd opened the pull request:
### Pull Request Description

- Closes #8722 

### Important Notes

<!--
- Mention important elements of the design.
- Mention any notable changes to APIs.
-->

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

radeusgd committed:
checkpoint

radeusgd committed:
implement and test ExternalLibrarySecretHelper

radeusgd committed:
safely decode secrets in AWS credentials, within a builder that does not expose them to Enso

radeusgd committed:
add tests for secrets in AWS cred

radeusgd committed:
fixes

radeusgd committed:
javafmt + fixes after rebase

jdunkerley commented on std-bits/aws/src/main/java/org/enso/aws/ClientBuilder.java:
```diff
@@ -0,0 +1,44 @@
+
+import org.enso.base.enso_cloud.ExternalLibrarySecretHelper;
+import org.enso.base.enso_cloud.HideableValue;
+import software.amazon.awssdk.auth.credentials.*;
```
can we avoid the * import please.

radeusgd replied:
Sure, somehow it got through `javafmt`

jdunkerley commented on std-bits/base/src/main/java/org/enso/base/enso_cloud/ExternalLibrarySecretHelper.java:
```diff
@@ -0,0 +1,57 @@
+   * will use to 'sign' their class name, proving that they can be trusted, without the need to
+   * update std-base whenever a new library is added.
+   */
+  private static void checkAccess() throws EnsoSecretAccessDenied {
```
do we know exactly how many steps up we need?
I think it is known to be up 2 frames.

radeusgd replied:
Yes, I think you are right. I will try to change this into a constant.

radeusgd committed:
CR1: *

radeusgd committed:
CR2: static frame offset

radeusgd committed:
fix

radeusgd committed:
increase timeouts

This is just a band-aid fix, we probably need to do something about the constant failures and heavy runner load, see: #8806

radeusgd committed:
retry

radeusgd committed:
scalafmt

radeusgd committed:
Merge branch 'develop' into wip/radeusgd/8722-secrets-in-aws-cred

radeusgd committed:
fix import...

hubertp commented on lib/scala/json-rpc-server-test/src/main/scala/org/enso/jsonrpc/test/JsonRpcServerTestKit.scala:
```diff
@@ -153,7 +153,7 @@ abstract class JsonRpcServerTestKit
     }
 
     def expectSomeJson(
-      timeout: FiniteDuration = 5.seconds.dilated
```
OK, let's revert in #8806 fix

radeusgd committed:
fix error log

radeusgd committed:
fix test path

radeusgd committed:
mark as flaky
